### PR TITLE
Make simple BlockBasedTableOptions mutable

### PR DESCRIPTION
**Pull request by pdillinger**
Summary: In theory, there should be no danger in mutability, as table
builders and readers work from copies of BlockBasedTableOptions.
However, there is currently an unresolved read-write race that
affecting SetOptions on BBTO fields. This should be generally
acceptable for non-pointer options of 64 bits or less, but a fix
is needed to make it mutability general here. See
https://github.com/facebook/rocksdb/issues/10079

This change systematically sets all of those "simple" options (and future
such options) as mutable. (Resurrecting this PR perhaps preferable to
proposed #13063)

Test Plan: Some unit test updates. XXX comment added to stress test code

### CODE REVIEW

**Review by ajkr — APPROVED** [review]
LGTM

**Comment by pdillinger** [comment]
Blocked on #10079

File: include/rocksdb/table.h [hunk 1]
```diff
@@ -125,7 +125,26 @@ struct CacheUsageOptions {
   std::map<CacheEntryRole, CacheEntryRoleOptions> options_overrides;
 };
 
-// For advanced user only
+// Configures how SST files using the block-based table format (standard)
+// are written and read.
+//
+// Except as specifically noted, all options here are "mutable" using
```
Comment (pdillinger) [inline include/rocksdb/table.h:131]: Oops, need to re-fix this comment

File: table/block_based/block_based_table_factory.cc [hunk 2]
```diff
@@ -396,36 +378,61 @@ static std::unordered_map<std::string, OptionTypeInfo>
...
+            // Values up to 64 bits should be generally safe to mutate despite
+            // the read-write race.
+            i.second.SetFlags(i.second.GetFlags() | OptionTypeFlags::kMutable);
+          }
+          break;
+      }
+    }
 
-};
+    // Exceptions to mutability:
```
Comment (cbi42) [inline table/block_based/block_based_table_factory.cc:424]: Is this line needed in case the default OptionTypeFlags can change to not kNone?

**Review by cbi42 — APPROVED** [review]
LGTM. Consider adding a change log.

File: table/block_based/block_based_table_factory.cc (hunk 2 above)
Comment (pdillinger) [reply table/block_based/block_based_table_factory.cc:424]: Oops, left over from previous code revision

**Pull request closed without merging**